### DISCUSSION TITLE
Hosting on AWS Lambda

### DISCUSSION THREAD

**Discussion by la-jamesh** [pocketbase/pocketbase#1276] [Q&A]
I'd like to bypass the server start and feed one request at a time into pocketbase, converting a lambda api gatway request into something pocket base understands. Is this something that's possible with the underlying API?

**Comment by ganigeorgiev** [comment]
I may misunderstood the question, but if you want to perform an API call against the PocketBase database without having the `serve` command running, then No - this is not possible, at least not out of the box.

You can create a custom console command (or even attach it to the `app.RootCmd` instance when [extending PocketBase](https://pocketbase.io/docs/use-as-framework/)) and inside it:
- start the server
- execute your API call (or if you extend PocketBase, you could also use directly [the DB and other helpers](https://pocketbase.io/docs/database/))
- end the server

PocketBase startup time is almost instantaneous so it kind of should work, but keep in mind that this way the [event hook](https://pocketbase.io/docs/event-hooks/) will fire unaware of the other "mini" PocketBase processes (aka. you'll have individual `core.App` instance for each request), meaning that the realtime subscriptions (and probably other things) will not work.
